test/python/circuit/test_hamiltonian_gate.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=arguments-differ,method-hidden
+
+"""Quick program to test the qi tools modules."""
+
+import json
+import numpy as np
+from numpy.testing import assert_allclose
+
+import qiskit
+from qiskit.extensions.hamiltonian_gate import HamiltonianGate
+from qiskit.test import QiskitTestCase
+from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.quantum_info import Operator
+from qiskit.transpiler import PassManager
+from qiskit.compiler import transpile
+from qiskit.converters import circuit_to_dag, dag_to_circuit
+from qiskit.transpiler.passes import CXCancellation
+
+
+class TestHamiltonianGate(QiskitTestCase):
+    """Tests for the HamiltonianGate class."""
+
+    def setUp(self):
+        """Setup."""
+        pass
+
+    def test_set_matrix(self):
+        """Test instantiation"""
+        try:
+            HamiltonianGate([[0, 1], [1, 0]], 1)
+        # pylint: disable=broad-except
+        except Exception as err:
+            self.fail('unexpected exception in init of HamiltonianGate: {0}'.format(err))
+
+    def test_set_matrix_raises(self):
+        """test non-unitary"""
+        try:
+            HamiltonianGate([[1, 0], [1, 1]], 1)
+        # pylint: disable=broad-except
+        except Exception:
+            pass
+        else:
+            self.fail('setting Unitary with non-hermitian matrix did not raise')
```

Comment:
You could use `assertRaises` here
```suggestion
        with self.assertRaises(ExtensionError):
            HamiltonianGate([[1, 0], [1, 1]], 1)
```